import_export/admin.py

```diff
@@ -489,6 +489,13 @@ def import_action(self, request, *args, **kwargs):
                     dataset = input_format.create_dataset(data)
                 except Exception as e:
                     self.add_data_read_fail_error_to_form(import_form, e)
+                else:
+                    if len(dataset) == 0:
```

Comment:
Would it be better to move this check to line 490 - i.e. before the except block?

Reply:
Not really. I think it's better to have a clean try block that only does what can raise the exception.